geo/src/types.rs

```diff
@@ -37,6 +37,16 @@ impl<F: GeoFloat> Closest<F> {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum GeoError {
```

Comment:
Calling this `GeoError` is a bit aspirational since they are only for boolean ops. Is it actually useful to distinguish between all these cases? (genuine question)

Reply:
yeah that isn't final yet 😅 I just figured it would be a good start to test it all out. I'm totally on board with you in creating a more specialized error enum there. The distinction between all the cases was made so that we would basically still have some rough information in the normal boolops code. Since we're just unwrapping there we will at least see the error enum variant which was chosen to mirror the previous error as close as possible.